src/NuGet.Core/NuGet.Commands/RestoreCommand/SourceRepositoryDependencyProvider.cs

```diff
@@ -346,7 +346,7 @@ public class SourceRepositoryDependencyProvider : IRemoteDependencyProvider
 
                 if (_throttle != null)
                 {
-                    await _throttle.WaitAsync();
+                    await _throttle.WaitAsync(cancellationToken);
```

Comment:
The linked issue makes it look like this PR is supposed to resolve a bunch of `CA2016` warnings (which our build would promote to errors). Shouldn't there also be a removal of `CA2016` from `NoWarn`, or `GlobalSuppressions.cs`?

Reply:
`CA2016` was never enabled in the repo and that's why IDE showed these warnings as `Info` messages (please see screenshot in the linked issue). Hence you don't see these warnings being suppressed in the `GlobalSuppressions.cs`.

Reply:
Then this PR should enable CA2016, so that after it's merged people don't introduce new violations?

Code changes should not only resolve the linked issue, but try to prevent regressions as well. That's what tests are for, in regards to changes in product behaviour. Therefore in the same spirit, these changes, which might not change product behaviour, need to consider other means to avoid future regressions.

Reply:
I enabled the rule in the ruleset and now is a `Warning`